app/controllers/site_controller.rb

```diff
@@ -428,9 +428,8 @@ def view_gene_expression
 
   # re-renders plots when changing cluster selection
   def render_gene_expression_plots
-    matches = @study.genes.by_name_or_id(params[:gene], @study.expression_matrix_files.map(&:id))
     subsample = params[:subsample].blank? ? nil : params[:subsample].to_i
```

Comment:
@study.expression_matrix_files.pluck(:id)  
:)

Reply:
The return value of `@study.expression_matrix_files` is actually an array, not a query, so that's why I'm using `map` here.